game/songs.cc

```diff
@@ -95,24 +94,8 @@ const std::string SONGS_CACHE_JSON_FILE = "songs.json";
 
 void Songs::LoadCache() {
 	const fs::path songsMetaFile = getCacheDir() / SONGS_CACHE_JSON_FILE;
-	std::ifstream file(songsMetaFile.string());
-	auto jsonRoot = nlohmann::json::array();
-	if (file) {
-		try {
-			std::stringstream buffer;
-			buffer << file.rdbuf();
-			file.close();
-			jsonRoot = nlohmann::json::parse(buffer);
-		} catch(std::exception const& e) {
-			std::clog << "songs/error: " << e.what() << std::endl;
-			file.close();
-			return;
-		}
-	} else {
-		std::clog << "songs/info: Could not open songs meta cache file " << songsMetaFile.string() << std::endl;
-		return;
-	}
-
+	auto jsonRoot = readJSON(songsMetaFile);
+	if (jsonRoot.empty()) return;
```

Comment:
Seems the `nlohmann/json.hpp` is still needed as `.empty()` isn't abstracted away

Reply:
Actually I think you're right; it was an oversight. It compiles fine on mac without the include.

Reply:
It should since this file includes `json.hh` which in turn includes this `nlohmann/json.hpp`, but then again having a file call something which is not directly listed as an include is nasty imho

Reply:
Yes, but IMO half the point of json.hh existing is not having to include nlohmann everywhere.